Mage.Sets/src/mage/cards/g/GluttonousHellkite.java

```diff
@@ -106,23 +106,22 @@ public boolean apply(Game game, Ability source) {
                 TargetSacrifice target = new TargetSacrifice(sacrificeAmount, StaticFilters.FILTER_PERMANENT_CREATURE);
                 if (target.canChoose(player.getId(), source, game)) {
                     player.choose(Outcome.Sacrifice, target, source, game);
-                    sacTargets.add(target);
+                    sacPermanents.add(game.getPermanent(target.getFirstTarget()));
                 }
             }
         }
-        for (TargetSacrifice sacTarget : sacTargets) {
-            for (UUID targetId : sacTarget.getTargets()) {
-                Permanent permanent = game.getPermanent(targetId);
-                if (permanent != null) {
-                    permanent.sacrifice(source, game);
+        for (Permanent permanent : sacPermanents) {
+            if (permanent != null) {
+                if (!permanent.sacrifice(source, game)) {
+                    sacPermanents.remove(permanent);
```

Comment:
okay, this has a subtle bug if the permanent is null. Better to null check it before adding it to the collection in the first place.

Also rather than removing to check the size, just increment a count.

Or can use stream logic:

```java
int counters = sacPermanents
    .stream()
    .mapToInt(p -> p.sacrifice(source, game) ? 2 : 0)
    .sum()
```

Reply:
I have updated to use this. Can you see a reason why a computer seems to be allowed to only sacrifice 1 creature when I have set X to 4 and I have to select 4 creatures?

Reply:
oh that sounds like #12044, use the workaround from SacrificeAllEffect:

```java
            while (!target.isChosen(game) && target.canChoose(player.getId(), source, game) && player.canRespond()) {
                player.choose(Outcome.Sacrifice, target, source, game);
            }
```